setup/plugins/installFiles/l10nDownload.civi-setup.php

```diff
@@ -0,0 +1,26 @@
+<?php
+/**
+ * @file
+ *
+ * Generate the civicrm.settings.php file.
+ */
+
+if (!defined('CIVI_SETUP')) {
+  exit("Installation plugins must only be loaded by the installer.\n");
+}
+
+/**
+ * Validate the $model.
+ */
+\Civi\Setup::dispatcher()
+  ->addListener('civi.setup.checkRequirements', function(\Civi\Setup\Event\CheckRequirementsEvent $e) {
+    $lang = \Civi\Setup::instance()->getModel()->lang;
+    if ($lang || $lang != 'en_US') {
+      $civicrmMoFileUrl = 'https://download.civicrm.org/civicrm-l10n-core/mo/' . $lang . '/civicrm.mo';
+      $dir = $e->getModel()->srcPath . DIRECTORY_SEPARATOR . 'l10n' . DIRECTORY_SEPARATOR . $lang . DIRECTORY_SEPARATOR . 'LC_MESSAGES';
+      \Civi\Setup::log()->info("Creating directry: " . $dir);
+      \CRM_Utils_File::createDir($dir, FALSE);
+      \Civi\Setup::log()->info("Download translation files from " . $civicrmMoFileUrl . ' into ' . $dir);
+      file_put_contents($dir . DIRECTORY_SEPARATOR . 'civicrm.mo', file_get_contents($civicrmMoFileUrl));
```

Comment:
Similarly I don't know how common it is to allow urls in file_get_contents(). Some sites might fail here. Guzzle is usually used for downloads now.

Reply:
I am now using Guzzle and have added some error detection.